### PR TITLE
feat(clerk-js,types): Add web3_coinbase_signature strategy on sign in/up flows

### PR DESCRIPTION
**Pull request by EmmanouelaPothitou**
## Description
Support a new strategy `web3_coinbase_signature` during sign in/up flows that uses the Coinbase Wallet.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Comment by clerk-cookie** [comment]
Hey @EmmanouelaPothitou - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 1.0.11-snapshot.vaadf3ba |
| @clerk/backend | 1.6.2-snapshot.vaadf3ba |
| @clerk/chrome-extension | 1.1.13-snapshot.vaadf3ba |
| @clerk/clerk-js | 5.14.0-snapshot.vaadf3ba |
| @clerk/elements | 0.12.4-snapshot.vaadf3ba |
| @clerk/clerk-expo | 2.1.0-snapshot.vaadf3ba |
| @clerk/express | 0.0.27-snapshot.vaadf3ba |
| @clerk/fastify | 1.0.29-snapshot.vaadf3ba |
| @clerk/localizations | 2.5.7-snapshot.vaadf3ba |
| @clerk/nextjs | 5.3.0-snapshot.vaadf3ba |
| @clerk/clerk-react | 5.4.0-snapshot.vaadf3ba |
| @clerk/remix | 4.2.13-snapshot.vaadf3ba |
| @clerk/clerk-sdk-node | 5.0.26-snapshot.vaadf3ba |
| @clerk/shared | 2.5.0-snapshot.vaadf3ba |
| @clerk/tanstack-start | 0.2.0-snapshot.vaadf3ba |
| @clerk/testing | 1.2.9-snapshot.vaadf3ba |
| @clerk/themes | 2.1.19-snapshot.vaadf3ba |
| @clerk/types | 4.13.0-snapshot.vaadf3ba |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@1.0.11-snapshot.vaadf3ba --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@1.6.2-snapshot.vaadf3ba --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.1.13-snapshot.vaadf3ba --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.14.0-snapshot.vaadf3ba --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.12.4-snapshot.vaadf3ba --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@2.1.0-snapshot.vaadf3ba --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@0.0.27-snapshot.vaadf3ba --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@1.0.29-snapshot.vaadf3ba --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@2.5.7-snapshot.vaadf3ba --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.3.0-snapshot.vaadf3ba --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.4.0-snapshot.vaadf3ba --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.13-snapshot.vaadf3ba --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.26-snapshot.vaadf3ba --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.5.0-snapshot.vaadf3ba --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.2.0-snapshot.vaadf3ba --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.2.9-snapshot.vaadf3ba --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.1.19-snapshot.vaadf3ba --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.13.0-snapshot.vaadf3ba --save-exact
```

**Comment by nikosdouvlis** [comment]
@EmmanouelaPothitou 👋🏻 
what's the status of this one?

File: packages/types/src/signUp.ts [hunk 1]
```diff
@@ -161,7 +162,7 @@ export type SignUpCreateParams = Partial<
 
 export type SignUpUpdateParams = SignUpCreateParams;
 
-export type SignUpAuthenticateWithMetamaskParams = {
+export type SignUpAuthenticateWithWeb3Params = {
```
Comment (chanioxaris) [inline packages/types/src/signUp.ts:165]: ⛔ This is a breaking change

File: packages/clerk-js/src/core/clerk.ts [hunk 2]
```diff
@@ -1366,6 +1367,53 @@ export class Clerk implements ClerkInterface {
     }
   };
 
+  public authenticateWithWeb3 = async ({
```
[the remark this answers is not in the document]
Comment (EmmanouelaPothitou) [reply packages/clerk-js/src/core/clerk.ts:1329]: I think we have 2 options here:

1. Deprecate authenticateWithMetamask and recommend transitioning to authenticateWithWeb3.
2. Retain authenticateWithMetamask, introduce authenticateWithCoinbase and optionally mark authenticateWithWeb3 as private.

**Pull request closed without merging**